### ISSUE TITLE
Unexpected tab order in calenader components

### ISSUE THREAD

**Issue by filiptammergard** [mantinedev/mantine#3660]
### What package has an issue

@mantine/dates

### Describe the bug

In v6, there is one tab stop for every day in the calendar. That means there are 38 tab stops just to tab past the component.

This is not following the APG guidelines for [date picker dialogs](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/):

> Note that, as specified in the [Grid Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/grid/), only one button in the calendar grid is in the Tab sequence.

The correct behavior can be seen in the native `<input type="date" />`: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date

It was also working as expected in v5: https://v5.mantine.dev/dates/calendar/#usage

### What version of @mantine/hooks page do you have in package.json?

6.0.0

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Change `tabIndex` logic to only put the selected day (or today/first day of displayed month if no day is selected) in tab order. All other days should have `tabIndex={-1}`.

**Comment by rtivital** [comment]
You are welcome to submit a PR to fix that in Calendar component, same logic should be applied to Moth and Year pickers. Please also include tests.
